compiler/resolution/functionResolution.cpp

```diff
@@ -9174,27 +9064,57 @@ static void replaceRuntimeTypePrims(std::vector<BaseAST*>& asts) {
 
 static Symbol* resolvePrimInitGetField(CallExpr* call);
 
-static Expr*   resolvePrimInit(CallExpr* call, Type* type);
+static void    resolvePrimInit(CallExpr* call, Symbol* val, Expr* fieldNameExpr, Type* type);
+static void    resolvePrimInit(CallExpr* call, Symbol* val, Type* type);
+
+static void resolvePrimInitNonGenericRecordVar(CallExpr* call,
+                                               Symbol* val,
+                                               AggregateType* at);
+
+static void resolvePrimInitGenericRecordVar(CallExpr* call,
+                                            Symbol* val,
+                                            AggregateType* at);
+
 
 static bool    primInitIsUnacceptableGeneric(CallExpr* call, Type* type);
 
 static void    primInitHaltForUnacceptableGeneric(CallExpr* call, Type* type);
 
-Expr* resolvePrimInit(CallExpr* call) {
-  Expr* retval = NULL;
-
-  if (SymExpr* se = toSymExpr(call->get(1))) {
+void resolvePrimInit(CallExpr* call) {
+  Expr* valExpr = NULL;
+  Expr* fieldNameExpr = NULL;
+  Expr* typeExpr = NULL;
+
+  INT_ASSERT(call->isPrimitive(PRIM_DEFAULT_INIT_VAR));
+
+  if (call->isPrimitive(PRIM_DEFAULT_INIT_VAR)) {
```

Comment:
Redundant given the assertion above. Could also remove the commented-out else branch.